### ISSUE TITLE
Can't cider-jack-in to projects with private git dependencies

### ISSUE THREAD

**Issue by PavlosMelissinos** [clojure-emacs/cider#3601]
## Expected behavior

cider-jack-in works in projects with private git repos

## Actual behavior

I'm getting some text in the nrepl-server buffer that asks for my passphrase but there's no interactive prompt, so I can't provide it to complete the process. Here's the exact message "prompt" as it appears in the nrepl buffer:

```
Cloning: git@github.com:org/private-repo-name.git

Enter passphrase for key '/home/pavlos/.ssh/id_ed25519': 
```

## Steps to reproduce the problem

Create a deps.edn with the following contents

```
{:deps {foo/bar {:git/url "git@github.com:org/private-repo-name.git"
                 :sha     "sha"}}}

```
where org/private-repo-name is a private repo

Run `M-x cider-jack-in`

### CIDER version information

```
;; CIDER 1.12.0 (Split), nREPL 1.0.0
;; Clojure 1.11.1, Java 21.0.1
```

### Lein / Clojure CLI version

```
Clojure CLI version 1.11.1.1149
```

### Emacs version

```
GNU Emacs 30.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.37, cairo version 1.16.0)
```

### Operating system

```
Fedora 39
```

### JDK distribution

```
openjdk version "21.0.1" 2023-10-17
OpenJDK Runtime Environment (Red_Hat-21.0.1.0.12-4) (build 21.0.1+12)
OpenJDK 64-Bit Server VM (Red_Hat-21.0.1.0.12-4) (build 21.0.1+12, mixed mode, sharing)
```

**Comment by PavlosMelissinos** [comment]
In case anyone else is facing this, here are some workarounds:

1. Spawn a standlone cider/nrepl server and cider-connect to it, [as described in the docs](https://docs.cider.mx/cider/basics/up_and_running.html#connect-to-a-running-nrepl-server)
2. Download the dependencies with `clojure -P` and then rerun cider-jack-in

**Comment by vemv** [comment]
Hi!

Thanks much for the report.

I'll try to repro it, anyway could you please provide the exact prompt text?

**Comment by PavlosMelissinos** [comment]
Good point, it's:

```
Cloning: git@github.com:org/private-repo-name.git

Enter passphrase for key '/home/pavlos/.ssh/id_ed25519': 
```

I've added it to the initial description too

**Comment by vemv** [comment]
Are you able to git-clone the same repo using e.g. Magit?

If not, it could indicate a generalized issue in your Emacs setup, such that `ssh-add` isn't being correctly invoked.

[Example](https://emacs.stackexchange.com/questions/41343/magit-asks-for-passphrase-for-ssh-key-every-time):

> [...] You can confirm this by running eval `$(ssh-agent); ssh-add ~/.ssh/id_rsa` in a terminal, and then in the same terminal `emacs &`. 

That solution isn't exactly ergonomic, but it would be start.

Please try that specific approach for CIDER and/or Magit, allowing us to discard hypotheses.

Thanks - V

**Comment by PavlosMelissinos** [comment]
> Are you able to git-clone the same repo using e.g. Magit?

Yes! If I delete the cache from .gitlibs and run magit-clone directly I get the correct prompt (same message but interactive)

**Comment by vemv** [comment]
Thanks!

To be clear, is your expected workflow entering the passphrase each time? It would make sense, although perhaps in your terminal emulator you don't have to?

**Comment by PavlosMelissinos** [comment]
> is your expected workflow entering the passphrase each time

I believe that depends on how the ssh agent is set up. So it doesn't happen always (I think) but when ssh-agent needs a passphrase I expect emacs/cider to ask for the user input correctly.

**Comment by PavlosMelissinos** [comment]
This use case no longer matters to me and there are no likes, so I'm closing it!
